ui/widgets/controlgroup.js

```diff
@@ -117,15 +117,17 @@ return $.widget( "ui.controlgroup", {
 				.each( function() {
 					var element = $( this );
 					var instance = element[ widget ]( "instance" );
+					var instanceOptions = $.widget.extend( {}, options );
 
 					// If the button is the child of a spinner ignore it
 					if ( widget === "button" && element.parent( ".ui-spinner" ).length ) {
 						return;
 					}
 					if ( instance ) {
-						options.classes = that._resolveClassesValues( options.classes, instance );
+						instanceOptions.classes =
+							that._resolveClassesValues( instanceOptions.classes, instance );
 					}
-					element[ widget ]( options );
+					element[ widget ]( instanceOptions );
```

Comment:
I can't be sure what the problem you're trying to fix is since you didn't include any description but im guessing its that when there are multiple of the same widget the options variable is over-written and re-used. if so i think we should fix this within the `_resolveClassesValues` function.